### PR TITLE
[Merged by Bors] - feat: port Data.Dfinsupp.Multiset

### PR DESCRIPTION
**Pull request by int-y1**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by int-y1** [comment]
Only `Multiset.equivDfinsupp` remains

**Comment by LukasMias** [comment]
Fixed it! What broke my brain is that trace.ext in mathlib3 did not give me the correct ext lemma to replace the broken ext with. I added that in a porting note:
```      /- potential bug: in lean 3, `trace.ext` outputs
         "matched goal to rule: dfinsupp.add_hom_ext'"
         but that lemma does not apply!-/```

At least my understanding was that whatever comes after "matched goal to rule" should be applicable to the current goal. But this did not work here. I do not currently have the energy to check if that is a known bug and/or make an issue.

File: Mathlib/Data/Dfinsupp/Multiset.lean [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      -- Porting note: used to be ext.
+      /- potential bug: in lean 3, `trace.ext` outputs
+         "matched goal to rule: dfinsupp.add_hom_ext'"
+         but that lemma does not apply!-/
+      apply Multiset.addHom_ext
+      simp)
+    (by
+      -- Porting note: used to be ext
+      apply Dfinsupp.add_hom_ext
+      simp)
```
Comment (dupuisf) [inline Mathlib/Data/Dfinsupp/Multiset.lean:97]:
```suggestion
      apply Dfinsupp.addHom_ext
      simp)
```
(Here of course I mean the name should be fixed wherever it is defined.)

**Review by dupuisf — COMMENTED** [review]
Thanks!

bors d+

File: Mathlib/Data/Dfinsupp/Multiset.lean (hunk 1 above)
Comment (int-y1) [reply Mathlib/Data/Dfinsupp/Multiset.lean]: done

**Pull request closed without merging**